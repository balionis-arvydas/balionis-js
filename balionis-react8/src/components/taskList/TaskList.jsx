import React from "react";

import { connect } from "react-redux";

import Task from "../task/Task";
import { mapStateToProps } from "./TaskListProperties";

import "./TaskList.less";

class TaskList extends React.Component {
    render () {
        const tasks = this.props.tasks.map(t => <Task key={t.id} task={t}/>);
        return (
            <div className="taskListWrapper">
                {tasks}
            </div>
        );
    }
}

export default connect(mapStateToProps)(TaskList);
