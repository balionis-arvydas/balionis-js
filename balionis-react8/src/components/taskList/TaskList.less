// @import "colors.less";
.taskListWrapper {
    border: 1px solid #0000ff;
}
