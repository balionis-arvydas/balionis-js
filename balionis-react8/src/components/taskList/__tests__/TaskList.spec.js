import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { mockStore } from '../../../redux/__tests__/mockStore';

import TaskList from '../TaskList';

describe('TaskList', () => {
    const tasks = [
        { id: 1, content: '12345' },
        { id: 2, content: '23456' },
    ];

    let store;

    beforeAll(() => {
        jest.resetAllMocks();
    });

    beforeEach(() => {
        store = mockStore({
            taskList: { tasks }
        });
        store.dispatch = jest.fn();
    });

    test("snapshot", async () => {
        const component = renderer.create(
            <Provider store={store}>
                <TaskList/>,
            </Provider>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

});
