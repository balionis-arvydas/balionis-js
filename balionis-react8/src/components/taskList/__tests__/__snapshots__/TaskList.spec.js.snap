// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList snapshot 1`] = `
Array [
  <div
    className="taskListWrapper"
  >
    <li
      className="taskWrapper"
      onClick={[Function]}
    >
      <span>
        12345
      </span>
    </li>
    <li
      className="taskWrapper"
      onClick={[Function]}
    >
      <span>
        23456
      </span>
    </li>
  </div>,
  ",",
]
`;
